web/modules/custom/dpl_related_content/src/Services/RelatedContent.php

```diff
@@ -0,0 +1,371 @@
+<?php
+
+namespace Drupal\dpl_related_content\Services;
+
+use Drupal\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Database\Database;
+use Drupal\Core\Datetime\DrupalDateTime;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Entity\FieldableEntityInterface;
+use Drupal\datetime\Plugin\Field\FieldType\DateTimeItemInterface;
+use Drupal\recurring_events\Entity\EventInstance;
+
+/**
+ * Load related nodes and events, based on filters.
+ */
+class RelatedContent {
+
+  /**
+   * The entity type interface.
+   */
+  protected EntityTypeManagerInterface $entityTypeManager;
+
+  /**
+   * The minimum of items that must be in the slider.
+   */
+  private int $minItems = 4;
+
+  /**
+   * How many items we max display in the slider.
+   */
+  private int $maxItems = 16;
+
+  /**
+   * The field on nodes, to sort by.
+   */
+  private string $nodeSortField = 'created';
```

Comment:
Consider making these class constants if they are not default values not intented to be changed.

Reply:
In a future PR, they will be overwritten (and set as public), so I'll keep them as properties.